beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -130,48 +132,27 @@ impl<T: BeaconChainTypes> DataAvailabilityChecker<T> {
         let current_epoch = current_slot.epoch(T::EthSpec::slots_per_epoch());
 
         if self.da_check_required_for_epoch(current_epoch) {
-            match availability_view.get_cached_block() {
+            match block {
                 Some(cached_block) => {
                     let block_commitments = cached_block.get_commitments();
-                    let blob_commitments = availability_view.get_cached_blobs();
 
                     let num_blobs_expected = block_commitments.len();
                     let mut blob_ids = Vec::with_capacity(num_blobs_expected);
 
                     // Zip here will always limit the number of iterations to the size of
                     // `block_commitment` because `blob_commitments` will always be populated
                     // with `Option` values up to `MAX_BLOBS_PER_BLOCK`.
-                    for (index, (block_commitment, blob_commitment_opt)) in block_commitments
-                        .into_iter()
-                        .zip(blob_commitments.iter())
-                        .enumerate()
+                    for (index, (_, blob_commitment_opt)) in
+                        block_commitments.into_iter().zip(blobs.iter()).enumerate()
                     {
                         // Always add a missing blob.
-                        let Some(blob_commitment) = blob_commitment_opt else {
+                        if blob_commitment_opt.is_none() {
                             blob_ids.push(BlobIdentifier {
                                 block_root,
                                 index: index as u64,
                             });
                             continue;
```

Comment:
`continue` is no longer required here

Reply:
true, and since we're not using `block_commitment` anymore maybe we get rid of this zip and make it more straightforward

```
                    let block_commitment_count = cached_block.get_commitments().len();

                    let blob_ids = blobs
                        .iter()
                        .take(block_commitment_count)
                        .enumerate()
                        .filter_map(|(index, blob_commitment_opt)| {
                            if blob_commitment_opt.is_none() {
                                Some(BlobIdentifier {
                                    block_root,
                                    index: index as u64,
                                })
                            } else {
                                None
                            }
                        })
                        .collect();
```